bigquery_etl/cli/stage.py

```diff
@@ -169,50 +175,59 @@ def deploy(
             artifact_file.rename(query_path)
             artifact_file = query_path
 
-        new_artifact_path = Path(sql_dir) / project_id / dataset / name
+        new_artifact_path = (
+            Path(sql_dir) / project_id / new_artifact_dataset / artifact_name
+        )
         new_artifact_path.mkdir(parents=True, exist_ok=True)
         shutil.copytree(artifact_file.parent, new_artifact_path, dirs_exist_ok=True)
         updated_artifact_files.add(new_artifact_path / artifact_file.name)
 
         # copy tests to the right structure
-        if test_path.exists():
-            test_destination = TEST_DIR / project_id / dataset / name
-            shutil.copytree(test_path, test_destination, dirs_exist_ok=True)
+        if artifact_test_path.exists():
+            new_artifact_test_path = (
+                TEST_DIR / project_id / new_artifact_dataset / artifact_name
+            )
+            shutil.copytree(
+                artifact_test_path, new_artifact_test_path, dirs_exist_ok=True
+            )
             if remove_updated_artifacts:
-                shutil.rmtree(test_path)
-
-            # rename test files
-            for test_file_path in map(Path, glob(f"{TEST_DIR}/**/*", recursive=True)):
-                for test_dep_file in artifact_files:
-                    test_project = test_dep_file.parent.parent.parent.name
-                    test_dataset = test_dep_file.parent.parent.name
-                    test_name = test_dep_file.parent.name
-
-                    file_suffix = test_file_path.suffix
-                    if test_file_path.name in (
-                        f"{test_project}.{test_dataset}.{test_name}{file_suffix}",
-                        f"{test_project}.{test_dataset}.{test_name}.schema{file_suffix}",
-                    ) or (
-                        test_file_path.name
-                        in (
-                            f"{test_dataset}.{test_name}{file_suffix}",
-                            f"{test_dataset}.{test_name}.schema{file_suffix}",
-                        )
-                        and test_project == project
-                    ):
-                        test_dataset = f"{test_dataset}_{project.replace('-', '_')}"
+                shutil.rmtree(artifact_test_path)
 
-                        if dataset_suffix:
-                            test_dataset = f"{test_dataset}_{dataset_suffix}"
-
-                        name = f"{project_id}.{test_dataset}.{test_name}"
-                        if test_file_path.name.endswith(f".schema{file_suffix}"):
-                            name += ".schema"
-                        name += file_suffix
+    # rename test files
+    for test_file_path in map(Path, glob(f"{TEST_DIR}/**/*", recursive=True)):
```

Comment:
Don't you only need to look at the files that were moved in the last step?  I know this wasn't changed in this pr but I think it's worth considering if I'm understanding correctly
```suggestion
    for test_file_path in map(Path, glob(f"{TEST_DIR}/{project_id}/**/*", recursive=True)):
```

Reply:
No, we need to look at all test files.  Only modified artifacts and upstream dependencies get deployed to stage, but any downstream dependencies of those have their SQL changed to refer to what got deployed to stage, and if those downstream dependencies have tests with data files meant to override the tables/views that did get deployed to stage then those need to get renamed to match.

For example, that's what's happening with #5446 to cause [three tests to fail](https://app.circleci.com/pipelines/github/mozilla/bigquery-etl/33104/workflows/e848735e-7a52-4ee4-9a7f-22334b2ce023/jobs/365287): the `events` views are getting deployed to stage, and `sql/moz-fx-data-shared-prod/fenix_derived/event_types_history_v1/query.sql` and `sql/moz-fx-data-shared-prod/fenix_derived/events_daily_v1/query.sql` are getting rewritten to refer to the staging versions of the `events` views, but the test data files in `tests/sql/moz-fx-data-shared-prod/fenix_derived/event_types_history_v1` and `tests/sql/moz-fx-data-shared-prod/fenix_derived/events_daily_v1` aren't getting renamed to match the staging versions of the `events` views.